src/randomeffectsterm.jl

```diff
@@ -53,3 +59,24 @@ function StatsModels.modelcols(t::RandomEffectsTerm, d::NamedTuple)
         z, z, LowerTriangular(Matrix{T}(I, S, S)), inds,
         sparse(II, J, vec(z)), Matrix{T}(undef, (S, length(invindex))))
 end
+
+
+# add some syntax to manually promote to full dummy coding
+fulldummy(t::AbstractTerm) =
+    throw(ArgumentError("can't promote $t (of type $(typeof(t))) to full dummy " *
+                        "coding (only CategoricalTerms)"))
+
+function fulldummy(t::CategoricalTerm)
+    new_contrasts = StatsModels.ContrastsMatrix(StatsModels.FullDummyCoding(),
+                                                t.contrasts.levels)
+    t = CategoricalTerm(t.sym, new_contrasts)
+end
+
+fulldummy(x) =
+    throw(ArgumentError("fulldummy can't be used outside of a mixedmodel formula"))
```

Comment:
Should this be `MixedModel` to refer to the package (and superclass) or `mixed-model` to refer to the concept?